domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/ApplicationLifecycleObserver.kt

```diff
@@ -83,6 +85,7 @@ class ApplicationLifecycleObserver @Inject constructor(
     ProcessLifecycleOwner.get().lifecycle.addObserver(this)
     application.registerActivityLifecycleCallbacks(this)
     logApplicationStartupMetrics()
+    logAllFeatureFlags()
```

Comment:
Please add tests to verify that this is happening & happening correctly.

Reply:
I have added a test to verify this behavior.